### PR TITLE
applications: nrf_desktop: Improve DFU over SMP documentation

### PR DESCRIPTION
**Pull request by MarekPieta**
Change updates documentation to inform about both modules used for the DFU over SMP. Change also contains other small improvements related to the SMP DFU documentation.

Jira: NCSDK-21667

### CODE REVIEW

File: applications/nrf_desktop/doc/ble_latency.rst [hunk 1]
```diff
@@ -51,10 +51,10 @@ The |ble_latency| uses delayed works (:c:struct:`k_work_delayable`) to control t
...
 
 * ``config_event`` - This event is received when the :ref:`nrf_desktop_config_channel` is in use.
-* ``ble_smp_transfer_event`` - This event is received when the firmware update is received by :ref:`nrf_desktop_ble_smp`.
+* ``ble_smp_transfer_event`` - This event is received when the firmware update is received either by :ref:`nrf_desktop_ble_smp` or :ref:`nrf_desktop_dfu_mcumgr`.
 
 When these events are received, the module sets the connection latency to low.
-When the :ref:`nrf_desktop_config_channel` is no longer in use and the firmware update is not received by :ref:`nrf_desktop_ble_smp` (no mentioned events for ``LOW_LATENCY_CHECK_PERIOD_MS``), the module sets the connection latency to :kconfig:option:`CONFIG_BT_PERIPHERAL_PREF_LATENCY`  to reduce the power consumption.
+When the :ref:`nrf_desktop_config_channel` is no longer in use and the firmware update is not received by :ref:`nrf_desktop_ble_smp` nor :ref:`nrf_desktop_dfu_mcumgr` (no mentioned events for ``LOW_LATENCY_CHECK_PERIOD_MS``), the module sets the connection latency to :kconfig:option:`CONFIG_BT_PERIPHERAL_PREF_LATENCY` to reduce the power consumption.
```
Comment (annwoj) [inline applications/nrf_desktop/doc/ble_latency.rst:57]:
```suggestion
When the :ref:`nrf_desktop_config_channel` is no longer in use, and neither :ref:`nrf_desktop_ble_smp` nor :ref:`nrf_desktop_dfu_mcumgr` receive firmware updates (no mentioned events for ``LOW_LATENCY_CHECK_PERIOD_MS``), the module sets the connection latency to :kconfig:option:`CONFIG_BT_PERIPHERAL_PREF_LATENCY` to reduce the power consumption.
```

File: applications/nrf_desktop/doc/dfu.rst [hunk 2]
```diff
@@ -102,7 +102,7 @@ The DFU module implementation is centered around the transmission and the storag
 * `Protocol operations`_ - How the module exchanges information with the host.
 * `Partition preparation`_ - How the module prepares for receiving an image.
 
-The firmware transfer operation can also be carried out by :ref:`nrf_desktop_ble_smp`.
+The firmware transfer operation can also be carried out either by :ref:`nrf_desktop_ble_smp` or :ref:`nrf_desktop_dfu_mcumgr`.
```
Comment (annwoj) [inline applications/nrf_desktop/doc/dfu.rst:105]:
```suggestion
The firmware transfer operation can also be carried out either by the :ref:`nrf_desktop_ble_smp` or :ref:`nrf_desktop_dfu_mcumgr`.
```

File: applications/nrf_desktop/doc/smp.rst [hunk 3]
```diff
@@ -29,4 +25,4 @@ Implementation details
 **********************
 
 nRF Desktop uses the |smp| from :ref:`lib_caf` (CAF).
-See the :ref:`CAF module <caf_ble_smp>` page for implementation details.
+See the :ref:`CAF module <caf_ble_smp>` page for implementation details and guide on how to perform the firmware update in the `nRF Connect for Mobile`_ application.
```
Comment (annwoj) [inline applications/nrf_desktop/doc/smp.rst:28]:
```suggestion
See the :ref:`CAF module <caf_ble_smp>` page for the implementation details and guide on how to perform the firmware update in the `nRF Connect for Mobile`_ application.
```

File: applications/nrf_desktop/README.rst [hunk 4]
```diff
@@ -1679,7 +1679,7 @@ The GATT Service is implemented by the :ref:`nrf_desktop_dev_descr`.
 Apart from the GATT Services, an nRF Desktop peripheral device must enable and configure the following application modules:
 
 * :ref:`nrf_desktop_ble_adv` - Controls the Bluetooth advertising.
-* :ref:`nrf_desktop_ble_latency` - Keeps the connection latency low when the :ref:`nrf_desktop_config_channel` is being used or when an update image is being received by the :ref:`nrf_desktop_ble_smp`.
+* :ref:`nrf_desktop_ble_latency` - Keeps the connection latency low when the :ref:`nrf_desktop_config_channel` is being used or when an update image is being received either by the :ref:`nrf_desktop_ble_smp` or :ref:`nrf_desktop_dfu_mcumgr`.
```
Comment (annwoj) [inline applications/nrf_desktop/README.rst:1682]:
```suggestion
* :ref:`nrf_desktop_ble_latency` - Keeps the connection latency low when the :ref:`nrf_desktop_config_channel` is being used or when either the :ref:`nrf_desktop_ble_smp` or :ref:`nrf_desktop_dfu_mcumgr` receive an update image.
```

**Review by annwoj — APPROVED** [review]

File: applications/nrf_desktop/README.rst [hunk 5]
```diff
@@ -1775,7 +1775,9 @@ The nRF Desktop application can use one of the following bootloaders:
     You can use the MCUboot for the background DFU through the :ref:`nrf_desktop_config_channel` and :ref:`nrf_desktop_dfu`.
     The MCUboot can also be used for the background DFU over Simple Management Protocol (SMP).
     The SMP can be used to transfer the new firmware image in the background, for example, from an Android device.
-    In that case, the :ref:`nrf_desktop_ble_smp` is used to handle the image transfer.
+    In that case, either the :ref:`nrf_desktop_ble_smp` or :ref:`nrf_desktop_dfu_mcumgr` is used to handle the image transfer.
+    The :ref:`nrf_desktop_ble_smp` relies on a generic module implemented in the Common Application Framework (CAF).
+    The :ref:`nrf_desktop_dfu_mcumgr` uses an application-specific implementation that allows to synchronize a secondary image slot access with the :ref:`nrf_desktop_dfu`.
```
Comment (kapi-no) [inline applications/nrf_desktop/README.rst:1780]: Consider mentioning the DFU lock utility:

```suggestion
    The :ref:`nrf_desktop_dfu_mcumgr` uses an application-specific implementation that allows to synchronize a secondary image slot access with the :ref:`nrf_desktop_dfu` using the :ref:`nrf_desktop_dfu_lock`.
```

**Review by mkapala-nordic — APPROVED** [review]

**Review by kapi-no — APPROVED** [review]

**Review by divipillai — APPROVED** [review]

**Pull request merged**